webui/react/src/hooks/useFetch.ts

```diff
@@ -81,6 +82,17 @@ export const useFetchUsers = (canceler: AbortController): () => Promise<void> =>
   }, [ canceler, storeDispatch ]);
 };
 
+export const useFetchUserSettings = (canceler: AbortController): () => Promise<void> => {
+  const storeDispatch = useStoreDispatch();
+
+  return useCallback(async (): Promise<void> => {
+    try {
+      const userSettingResponse = await getUserSetting({}, { signal: canceler.signal });
```

Comment:
This will definitely be an improvement over what currently exists but I wanted to call out a few issues we should address in the future:
1. Right now this initial fetch is always made. We should avoid a fetch if the store is already up-to-date.
2. Maybe this is buried in the auto-generated gRPC code but we aren't de-duplicating requests. if another call to `useFetchUsersSettings(...)()` is made before the first one is complete, even if point 1 is addressed we still make another call.